content/blog/2021-11-17-exp-versioning.md

```diff
@@ -0,0 +1,300 @@
+see the benefits of this approach.
+
+# Benefits of Versioning
+
+With an experiment tracking API, you insert API calls in your code to save
+experiment information in a central database. With experiment versioning,
+experiment information lives in readable files that are always available, and
+your code stays clean.
+
+Even without DVC, you can read, save, and version your experiment parameters and
+metrics:
+
+```dvc
+$ cat params.yaml
+train:
+  epochs: 10
+model:
+  conv_units: 128
+```
+
+```dvc
+$ cat metrics.json
+{"loss": 0.24310708045959473, "acc": 0.9182999730110168}
+```
+
+You can see what changed in parameters, code, or anything else:
+
+```dvc
+$ git diff HEAD~1 -- params.yaml
+diff --git a/params.yaml b/params.yaml
+index baad571a2..57d098495 100644
+--- a/params.yaml
++++ b/params.yaml
+@@ -1,5 +1,5 @@
+ train:
+   epochs: 10
+-model:
+-  conv_units: 16
++model:
++  conv_units: 128
+```
+
+Many experiment tracking databases do not capture everything you need to
+reproduce an experiment, especially the pipeline to run the experiment end to
+end. With experiment versioning, it's important to codify your experiment
+pipeline so that anyone can reproduce the experiment. Notice that the
+`dvc exp init` command above generated a `dvc.yaml` file that describes the
+experiment pipeline:
+
+```dvc
+$ cat dvc.yaml
+stages:
+  default:
+    cmd: python src/train.py
+    deps:
+    - data/images
+    - src/train.py
+    params:
+    - model
+    - train
+    outs:
+    - models
+    metrics:
+    - metrics.json:
+        cache: false
+    plots:
+    - logs.csv:
+        cache: false
+```
+
+This captures:
+
+- The command to run the experiment.
+- Parameters and other dependencies.
+- Metrics, plots, and other outputs.
+
+The pipeline has a single `default` stage, but more stages can be added.
+
+With an experiment tracking dashboard, you see all your experiments, and you may
+sort, label, and filter to help keep them organized. With experiment versioning,
+your code and experiments stay connected, and you have more flexibility in how
+you organize and share them.
+
+For example, you may try an experiment in a new Git branch. If something goes
+wrong or the results are uninspiring, you may not push the branch, in which case
+it doesn't get shared, reducing unnecessary clutter and cleanup that you might
+encounter in an experiment tracking database and dashboard.
+
+If the experiment instead looks interesting, you can push it along with your
+code so that they stay in sync and are shared with the same people, and it's
+already organized using your existing branch name. You can keep iterating on
+that branch, start a new one if an experiment diverges too much, or merge into
+your main branch to make it your primary model.
+
+# Benefits of DVC
+
+Of course, you could do all this yourself in Git. Since we have established that
+code version control is insufficient for experiments, DVC adds features specific
```

Comment:
The benefits you list are pretty narrowly focused on experiment organization and miss other benefits like reproducibility. Do we want to mention things like data management, caching, and pipeline execution that are major benefits of DVC, and are hard to do with either Git or experiment trackers by themselves? Where would you put this?

Reply:
> miss other benefits like reproducibility.

💯 

> data management, caching, and pipeline execution that are major benefits of DVC

not an easy question. It would be great to have these but it will move away the user focus from experiment versioning. It would be great to mention this but not spend too much time on this.